### PR TITLE
Documentation III

### LINKED ISSUE #121: Document steps for how to publish a uniffi'd project on npmjs
After setting up UBRN in a new project created with create-react-native-library@0.35.1, I had to add the following to `.gitignore` to exclude the Rust checkouts and binaries from git:

```
# uniffi-bindgen-react-native
rust_modules/
*.a
```

When packing the library for publishing, npm appears to have special logic for combining `.gitignore` and the `files` array from `package.json`.

For Android, `*.a` in `.gitignore` and `"android"` in `files` results in the `android/src/main/jniLibs/$target/$library.a` files being included in the package.

For iOS, however, in order to include `build/$library.xcframework`, I had to add `"build"` to `files` to undo the `build/` from `.gitignore`.

I'm not yet sure how to best publish packages built with UBRN but decided to log this here for documentation in the meanwhile. Changes to `.gitignore` and or `files` appear to be needed regardless of whether one wants to include the binaries in the package or not.

### PR DESCRIPTION
**Pull request by jhugman**
Fixes #121 

This PR completes the baseline documentation (I think).

It adds:

- internals - sketching:
  -  the codegen'd path that the `generate turbo-modules` uses
  - the lifting and lowering path that the generated bindings uses
- callback interfaces and async-callbacks
- contribution guide for documentation.
- miscellaneous typo fixes and refactoring

Reviewing theses docs has [a guide now](https://github.com/jhugman/uniffi-bindgen-react-native/pull/125/files#diff-41597838bd197cee183fc1e42dfa9303dcccbef48718c6085c94962fc18574dd)! 

@Johennes : this page could do with [your review/input](https://github.com/jhugman/uniffi-bindgen-react-native/pull/125/files#diff-51fd45086a4403854ace39809b97012ec40ba309f538dae986ebdb5d2d22ae99).

### CODE REVIEW

File: docs/src/idioms/async-callbacks.md [hunk 1]
```diff
@@ -1 +1,49 @@
...
+
+You can see this in action in the [`futures` fixture](https://github.com/jhugman/uniffi-bindgen-react-native/tree/main/fixtures/futures).
+
+## Task cancellation
+
+When the Rust Future is completed, it is dropped, and Typescript is informed. If the Future is dropped before it has completed, it has been cancelled. `uniffi-bindgen-react-native` can use this information to call the async callback to cancel, using [the standard `AbortController` and `AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) machinery.
+
+`uniffi-bindgen-react-native` generates an optional argument for each async callback method, which is an options bag containing an `AbortSignal`.
+
+It is up to implementer of each method whether they want to use it or not.
```
Comment (Johennes) [inline docs/src/idioms/async-callbacks.md:37]:
```suggestion
It is up to the implementer of each method whether they want to use it or not.
```

File: docs/src/guides/getting-started.md [hunk 2]
```diff
@@ -136,7 +135,6 @@ Until then, you need to add the dependency to the app's Podfile, in this case `e
 +  # We need to specify this here in the app because we can't add a local dependency within
```
Comment (zzorba) [inline docs/src/guides/getting-started.md:135]: Is publishing to cocoapods still an option?

File: docs/src/idioms/callback-interfaces.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+If an error is thrown in Typescript, it ends up in Rust:
+
+```typescript
+class ConsoleLogger implements MyLogger {
+    isEnabled(): boolean {
+        return false;
+    }
+    log(message: string) {
+        if (!this.isEnabled()) {
+            throw new MyError.LoggingDisabled();
```
Comment (zzorba) [inline docs/src/idioms/callback-interfaces.md:90]: What happens if I throw an error that doesn't conform to the MyError declared in the rust code?

**Review by zzorba — APPROVED** [review]
LGTM, one small question/clarification

File: docs/src/idioms/callback-interfaces.md (hunk 3 above)
Comment (jhugman) [reply docs/src/idioms/callback-interfaces.md:90]: Aha. I'd missed this. Thanks for spotting that.

I've added some more docs.

**Pull request merged**